server/service/src/asset/insert.rs

```diff
@@ -38,7 +41,7 @@ pub fn insert_asset(
         .connection
         .transaction_sync(|connection| {
             validate(&input, connection)?;
```

Comment:
I think we need to validate that the location isn't already used by another asset.
Ideally we should warn the user and maybe remove the location from the other asset if they really want to do it.
Not sure the best way to alert the user but for now we should at least error if they try to use a location already assigned to another asset.

Reply:
Curious! I think I wasn't understanding how assets and locations worked.

Do Locations sit within Assets? Or are Assets placed at Locations?

Reply:
Locations are within the asset. Location name might be `Fridge 1 - Shelf 3` or something similar.
I tried to communicate that by calling it `asset_internal_location` to differentiate it from where an asset might be.
If we do other assets say Oxygen tanks in future, they might be stored in a location, which will require a different approach.

Hope that makes sense?

Reply:
Yes makes sense, thanks James